tests/vm_delete_protection_test.go

```diff
@@ -29,7 +29,6 @@ var _ = Describe("VM delete protection", func() {
 		if vm != nil {
 			err := apiClient.Get(ctx, client.ObjectKeyFromObject(vm), vm)
 			Expect(err).To(Or(Not(HaveOccurred()), MatchError(errors.IsNotFound, "errors.IsNotFound")))
-
```

Comment:
Please also add a check for the deletion timestamp. If it is there, skip the rest of the code.

Reply:
Done.